lib/sqs.js

```diff
@@ -72,13 +154,47 @@ module.exports = ({ account, queuePrefix = '', queueSuffix = '', defaultVisibili
         }
       }
 
-      return SQS.sendMessageAsync({
-        MessageBody: payload,
-        QueueUrl: getQueueURL(queueName),
-        MessageAttributes: attrs
-      });
+      let key = false;
+      const messageStream = compress(payload);
+
+      if (Buffer.byteLength(payload) > TWO_FIFTYSIX_K) {
```

Comment:
think this logic is incorrect, we need to make sure that the message isn't too large _after_ base64 encoding it right?

Also I think attributes play a part in the size calculation too FWIW

Reply:
good call on attributes, didn't realize that.  any good ideas on how to do that calculation?  is stringifying the attributes object and adding that size enough or ...?

Reply:
yea honestly not 100% sure.  save ourselves 1K for attributes and hope?  (that is exactly the kind of suggestion that's going to burn us down the road though)